exercises/grep/grep.spec.js

```diff
@@ -0,0 +1,371 @@
+const { spawnSync } = require("child_process");
+const path = require("path");
+
+const BASE_DIR = path.resolve(__dirname);
+
+const resolveDataFile = file => path.resolve(BASE_DIR, "data", file);
+
+const spawn_grep = config => {
+  const args = [
+    path.resolve(BASE_DIR, "grep.js"),
+    ...config.flags,
+    config.pattern,
+    ...config.files.map(file => path.resolve(BASE_DIR, "data", file))
+  ];
+
+  return new Promise((resolve, reject) => {
+    const child = spawnSync("node", args, { stdio: "pipe" });
+    const stderr = child.stderr.toString().trim();
+    const stdout = child.stdout.toString().trim();
+
+    if (stderr) {
+      reject(stderr);
```

Comment:
Shouldn't here be a return as well?
Else it would try also to run `resolve(stdout)` below, right?

Reply:
I've done a few tests, it seems to work fine. I added an else statement though because it's less error prone indeed

Reply:
Sounds like a good idea. Promises cannot be changed, but I was not quite sure about the implementation. I think I also had one once where it would throw if you try to call resolve or reject a second time.